pulp_ansible/app/utils.py

```diff
@@ -26,3 +27,26 @@ def filter_content_for_repo_version(qs, repo_version):
     content_rel = RepositoryContent.objects.filter(f)
 
     return qs.filter(pk__in=content_rel.values_list("content_id"))
+
+
+def get_queryset_annotated_with_task_sync_task(qs):
+    tasks = (
+        Task.objects.filter(
+            name__contains="sync", reserved_resources_record__icontains=OuterRef("outer_pk")
+        )
+        .annotate(
+            task=JSONObject(
+                pk=F("pk"),
+                state=F("state"),
+                pulp_created=F("pulp_created"),
+                finished_at=F("finished_at"),
+                error=F("error"),
+            )
+        )
```

Comment:
Why is this needed? Can you just use `only()`?

Reply:
I need the `JSONObject` so I can annotate more than one field, made it a slightly better with the latest solution.